File: 515. Find Largest Value in Each Tree Row.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
let largestValues = function(root) {
    let output = []
    let queue = []
    
    if (!root) {
        return []
    }

    if (root == null) {
        return root
    }
    
    queue.push(root)
    output.push(root.val)
    
    while (queue.length > 0) {
        let level = []
        let nums = []
        
        while (queue.length > 0) {
            let node = queue.shift()
            
            if (node.left) {
                level.push(node.left)
                nums.push(node.left.val)
            }
            if (node.right) {
                level.push(node.right)
                nums.push(node.right.val)
            }
        }
        
        if (nums.length > 0) {
            let big = Math.max.apply(Math, nums)
            output.push(big)
        }
        queue = level
    }
    
    return output
};

File: 515. Find Largest Value in Each Tree Row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        output = []
        
        if not root:
            return output
    
        q1 = [root]

        while q1:
            q2 = []
            output.append(max([node.val for node in q1]))

            for node in q1:
                if node.left:
                    q2.append(node.left)
                if node.right:
                    q2.append(node.right)

            q1 = q2
   
        return output
